gst/nnstreamer/tensor_query/tensor_query_serversink.c

```diff
@@ -310,20 +248,42 @@ gst_tensor_query_serversink_render (GstBaseSink * bsink, GstBuffer * buf)
 {
   GstTensorQueryServerSink *sink = GST_TENSOR_QUERY_SERVERSINK (bsink);
   GstMetaQuery *meta_query;
-  query_connection_handle conn;
+  nns_edge_data_h data_h = NULL;
+  guint i, num_mems = 0;
+  gint ret;
+  GstMemory *mem[NNS_TENSOR_SIZE_LIMIT];
+  GstMapInfo map[NNS_TENSOR_SIZE_LIMIT];
+  char *val;
 
   meta_query = gst_buffer_get_meta_query (buf);
   if (meta_query) {
     sink->metaless_frame_count = 0;
-    conn = nnstreamer_query_server_accept (sink->server_data,
-        meta_query->client_id);
-    if (conn) {
-      if (!tensor_query_send_buffer (conn, GST_ELEMENT (sink), buf)) {
-        nns_logw ("Failed to send buffer to client, drop current buffer.");
+
+    ret = nns_edge_data_create (&data_h);
+    if (ret != NNS_EDGE_ERROR_NONE) {
+      nns_loge ("Failed to create data handle in server sink.");
+      return GST_FLOW_ERROR;
+    }
+
+    num_mems = gst_buffer_n_memory (buf);
+    for (i = 0; i < num_mems; i++) {
+      mem[i] = gst_buffer_peek_memory (buf, i);
+      if (!gst_memory_map (mem[i], &map[i], GST_MAP_READ)) {
+        ml_loge ("Cannot map the %uth memory in gst-buffer.", i);
+        num_mems = i;
+        goto done;
       }
-    } else {
-      nns_logw ("Cannot get the client connection, drop current buffer.");
+      nns_edge_data_add (data_h, _g_memdup (map[i].data, map[i].size),
```

Comment:
IMO copying memory is useless here. (remove free function and set mapped data)

Reply:
Updaed.